src/common/catalog/catalog-category-registry.ts

```diff
@@ -19,26 +19,38 @@
  * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 
-import { action, computed, observable, toJS } from "mobx";
+import { action, computed, observable } from "mobx";
+import { Disposer, ExtendedMap } from "../utils";
 import { CatalogCategory, CatalogEntityData, CatalogEntityKindData } from "./catalog-entity";
 
 export class CatalogCategoryRegistry {
-  @observable protected categories: CatalogCategory[] = [];
+  protected categories = observable.set<CatalogCategory>();
 
-  @action add(category: CatalogCategory) {
-    this.categories.push(category);
+  @action add(category: CatalogCategory): Disposer {
+    this.categories.add(category);
+
+    return () => this.categories.delete(category);
   }
 
-  @action remove(category: CatalogCategory) {
```

Comment:
@Nokel81  was this removed by accident?

Reply:
Maybe not, would have been good to mention that we are breaking the api here ... 🙂

Reply:
Sorry, will start listing breaking API changes in my PR descriptions from now on.